homeassistant/components/mochad.py

```diff
@@ -34,7 +35,8 @@
 }, extra=vol.ALLOW_EXTRA)
 
 
-def setup(hass, config):
+@asyncio.coroutine
+def async_setup(hass, config):
```

Comment:
Good point. There is an I/O to instantiate pymochad where connection is made. This is what I want to make non-blocking for the homeassistant. Instead of async, let me re-work this to MochadCtrl(threading.Thread)  and move the Pymochad() to the def run() method. This way def setup(hass, config) will quickly create a thread and return.

Reply:
There's no big gain in making the setup function async if all other calls are done in threads.